### PR TITLE
deps: bump eslint and related items

### PR DESCRIPTION
**Pull request by paulirish**
the 4 commits each have a unique thing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,10 +146,11 @@
     "csv-validator": "^0.0.3",
     "devtools-protocol": "0.0.939404",
     "es-main": "^1.0.2",
-    "eslint": "^7.23.0",
-    "eslint-config-google": "^0.9.1",
-    "eslint-plugin-import": "^2.24.2",
-    "eslint-plugin-local-rules": "0.1.0",
+    "eslint": "^8.4.1",
+    "eslint-config-google": "^0.14.0",
+    "eslint-formatter-codeframe": "^7.32.1",
```
Comment (brendankenny) [inline package.json:151]: what does this do?

Comment (paulirish) [reply package.json:151]: image

we have it configured so the codeframe formatter is used in CI so if it fails there we see exactly what the prob is in the logs.

i also prefer using it locally. 

in latest eslint they extracted the formatter so its not shipped in the main pkg anymore.

File: lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js [hunk 2]
```diff
@@ -198,8 +198,8 @@ class NetworkAnalyzer {
       if (!Number.isFinite(timing.receiveHeadersEnd) || timing.receiveHeadersEnd < 0) return;
       if (!record.resourceType) return;
 
-      const serverResponseTimePercentage = SERVER_RESPONSE_PERCENTAGE_OF_TTFB[record.resourceType]
-        || DEFAULT_SERVER_RESPONSE_PERCENTAGE;
+      const serverResponseTimePercentage =
+      SERVER_RESPONSE_PERCENTAGE_OF_TTFB[record.resourceType] || DEFAULT_SERVER_RESPONSE_PERCENTAGE;
```
Comment (adamraine) [inline lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js:202]:
```suggestion
      const serverResponseTimePercentage =
        SERVER_RESPONSE_PERCENTAGE_OF_TTFB[record.resourceType] || DEFAULT_SERVER_RESPONSE_PERCENTAGE;
```

Comment (paulirish) [reply lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js]: heh yeah i had touched this one manually but doing that change goes over 100chars :) 

anyway, just added another break after the `||` to sort it, along with those indents.

**Review by adamraine — APPROVED** [review]
LGTM

**Pull request merged**